src/inmanta/data/dataview.py

```diff
@@ -139,6 +155,21 @@ def has_start(self) -> bool:
     def has_end(self) -> bool:
         return (self.end is not None) or (self.last_id is not None)
 
+    def get_empty_page_boundaries(self) -> PagingBoundaries:
+        """
+        Return the virtual paging boundaries that corresponds to the boundaries of an empty page response for this request.
+
+        Concretely, this means swapping start and end, because the semantics of PagingBoundaries are opposite those of this
+        class: PagingBoundaries represents "current page" while this class represents "requested page";
+        max of current page = min of requested page and vice versa.
+        """
+        return PagingBoundaries(
```

Comment:
Perhaps we should add a comment here noting that there is a potential off-by-one error here because both classes are exclusive of the boundary values, and add a reference to the remaining ticket?